### ISSUE TITLE
USB4VC and KVM?

### ISSUE THREAD

**Issue by vorvek** [dekuNukem/USB4VC#3]
Hello, I'm trying to use my USB4VC+IBM board with an Aten PS\2+VGA KVM. The results are not good; in my slowest computers I can type a dozen characters before I get a beep from the PC speaker. It doesn't matter if I type them fast or slow, sometimes writing a simple command in DOS will require me to wait for a second. All the letters appear afterwards, however. In faster machines (ie, Pentium and beyond) it seems the mouse is more problematic, although some of my keyboard inputs are also eaten.

Am I correct to assume that the issue comes from having a KVM in between? I can give it a try, connecting the computers directly, but the back of my rack is somewhat inaccessible (and those PS\2 connectors love to get their pins bent when you try to plug them blind).

**Comment by dekuNukem** [comment]
Hi there! Please do try it directly attached to the computer if possible, and let me know how it goes.

I did all the developments with computers and it seem to work fine, so maybe the KVM is doing something funky.

Is this the model you're using?

https://www.amazon.co.uk/Aten-CS22U-2-Port-Cable-Switch-black/dp/B002NTIZCG

**Comment by vorvek** [comment]
No, it's a rack mounted KVM with 8 ports, one of these: https://www.kvm-switches-online.com/cs9138.html

I'll try connecting the stuff directly later this week and report back!

**Comment by vorvek** [comment]
I've tried connecting directly to a few of the computers and this does make the USB4VC work in what I can only call flawless fashion. I've also tested the KVM I was using previously, a 4 port Aten Master View 114-A which from my understanding is a consumer version of the one I'm currently using. By this, I mean that the innards are very similar except for a lack of OSD, but it also has the PS\2 emulation aspect and power through VGA. 

The results are the same as with the CS9138, stuttery mouse and keyboard and weird input handling, although I've only tested it in a couple of the computers (I run my VGA output through an OSSC, it's really a pain in the ass to unplug some of this stuff, but it's much worse to plug it later!).

While I can't part with my current 8 port for a long period of time, I can mail you the 4 port KVM. I'm also in the UK (London, thus the tiny computer room lol), I don't mind paying for the shipping, and you can use it for any tests you consider relevant. I don't have any other KVMs to test the USB4VC with, but I assume most VGA+PS\2 KVMs will be pretty much the same, particularly if they are "proper" products from the era when they were relevant.

**Comment by dekuNukem** [comment]
Good to hear that it works fine on the computer itself! Really curious to see what shenanigans the KVM is doing.

Thank you very much for offering the KVM for me to test! I’m happy to cover the postage, although I might need to have it for a week or two, so hopefully you don't depend on it too much. Otherwise I might just buy one myself.

Anyway, feel free to let me know your email or [DM me on discord](https://discord.gg/HAuuh3pAmB), so we can exchange the information in private.

**Comment by dekuNukem** [comment]
Hi there! Thank you very much for sending me the KVM to check it out. It helped immensely with adding and testing host inhibition, resulting in improved compatibility with different systems. 

I feel that I'm pretty much done with it now, would you like to have it back? I can hold on to it if you’re not in a hurry, but otherwise I can post it back anytime you want!

**Comment by vorvek** [comment]
Not in a hurry to get it back, feel free to send it back whenever you see fit. The return address was in the wrapping, so I assume you don't have it anymore, haha. Let me know in Discord when you want to post it and I'll give you the address.

**Comment by queenkjuul** [comment]
Just want to chime in and say I am also having issues with a KVM - a couple in fact. Haven't had time yet to do meaningful testing. 

I have a USB KB -> USB KVM -> USB4VC -> PS/2 KVM setup. Notably, the USB4VC screen indicates that it is detecting multiple keyboards attached. I was experiencing similar issues - either big lag, or key repeats, or key drops, plus completely erratic mouse behavior. Will do some testing with each KVM individually when I can. Also just got a new PS/2 KVM to try as well.

**Comment by ackmangogo** [comment]
Aten KVM as I know will emulate ps/2 mouse and keyboard, maybe passive ps/2 switch will help in this use case.

**Comment by Jimbo7136** [comment]
I am using an ATEN CS84U PS2/USB KVM after the USB4VC. The keyboard works but the mouse doesn't at all. Plugging the PS2 ports directly from the USB4VC to a PC works correctly.

**Comment by schlangz** [comment]
I have the same problem, using DIGITUS KVM Switch DC IC-814IO, mouse is not working at all. Plugging the PS2 cable directly into the PC works. Any way to debug this?

**Comment by jbbbloody** [comment]
Just noticed this and wanted to add, that I am using the USV4VC with an old rack-mountable COMPAQ 8 Port VGA & PS/2 KVM with OSD and its working like a charm. Even using it with all my modern Kit, just depends on the PS/2 to USB converters you attach, some are causing double characters, or missing input... but in general they work like a charm.
